data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -532,10 +536,11 @@ public static JacksonEvent fromEvent(final Event event) {
     public abstract static class Builder<T extends Builder<T>> {
 
         private EventMetadata eventMetadata;
-        private Object data;
+        protected Object data;
```

Comment:
Why do you need to expose this further?

If you do need it, you should have a protected getter: `protected Object getData()`.

Reply:
Yeah, I do need it because I am exposing it to derived classes.

Reply:
Actually, I need it so that I can modify it directly from derived classes, getData() does not work. I can add `setData()` but I think it makes sense to actually use `protected` member so that derived classes can directly modify it. I think it makes sense in this case. Let me know what you think. The current code is broken in the sense that `JacksonMetric` has it's own `data` which is different from parent class's `data` and that is problematic in some cases.